### DISCUSSION TITLE
[Core Protocol] Questions about making op-node stateful and the custom p2p messages in op-geth

### DISCUSSION THREAD

**Discussion by lesterli** [ethereum-optimism/developers#408] [Core Protocol]
### Did you check the documentation?

- [X] Yes

### Did you read the specs?

- [ ] Yes
- [ ] No

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

We are a startup working on a project that requires the sequencer to store some additional data off-chain. We are considering adding an RPC API within `op-node` to store these data in a PostgreSQL database. But we have two concerns:
1. This turns `op-node` into a stateful service. Is it architecturally sound to do so within an OP stack for its operation and scalability?
2. Furthermore, there will be another RPC for querying these data, so we desire to broadcast this non-block/transaction data to other op-nodes via the P2P network to keep the nodes in sync. Maybe I need to modify [the protocol code](https://github.com/ethereum-optimism/op-geth/blob/optimism/eth/protocols/eth/protocol.go) in `op-geth` to support the broadcasting of custom messages, do you have any suggestions about this?

**Comment by opfocus** [comment]
@sbvegan
